### PR TITLE
Fix dispatch for array that disallows missing

### LINKED ISSUE #112: Allow dims arg to Filter `apply` when there are no missings
Use cases for `Filter` may be agnostic to whether the data allows missing values, and therefore they will pass the `dims` kwarg.

For example:

```julia
using Impute

function my_apply(data::AbstractArray, f::Filter; dims)
    # data may or may not allow/have missing values
    Impute.apply(data, f; dims=dims)
end
```

If `data` does not allow missing, the `apply` will dispatch to the [table method](https://github.com/invenia/Impute.jl/blob/70ccac3e6bb861015b113ac114ffbd9ed5c1d2e6/src/filter.jl#L44) and throw a `MethodError`.

To get the [appropriate method](https://github.com/invenia/Impute.jl/blob/70ccac3e6bb861015b113ac114ffbd9ed5c1d2e6/src/filter.jl#L42), we therefore need to add `kwargs...` to it.

### PR DESCRIPTION
**Pull request by bencottier**
Closes #112

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name = "Impute"
 uuid = "f7bf1975-0170-51b9-8c5f-a992d46b9575"
 authors = ["Invenia Technical Computing"]
-version = "0.6.3"
+version = "0.6.4"
```
Comment (bencottier) [inline Project.toml:4]: Not sure if considered a patch or a feature

Comment (rofinn) [reply Project.toml:4]: No, I think this is just a bug / edge cases. The corresponding `impute` method already does this.

**Review by rofinn — APPROVED** [review]
LGTM

**Comment by bencottier** [comment]
I'm not authorized to merge.

**Pull request merged**